tidy3d/components/medium.py

```diff
@@ -2581,8 +2582,11 @@ def _derivative_field_cmp(
 
         # TODO: probably this could be more robust. eg if the DataArray has weird edge cases
         E_der_dim = E_der_map[f"E{dim}"]
-        E_der_dim_interp = E_der_dim.interp(**coords_interp).fillna(0.0).sum(dims_sum)
-        vjp_array = np.array(E_der_dim_interp.values).astype(complex)
+        E_der_dim_interp = E_der_dim.interp(**coords_interp).fillna(0.0).sum(dims_sum).real
+        E_der_dim_interp = E_der_dim_interp.sum("f")
+
+        vjp_array = np.array(E_der_dim_interp.values).astype(float)
```

Comment:
Was the `.astype(complex)` here wrong? Wondering if the different return type here affects other things.

Reply:
not sure it was wrong, I just did the real part later. basically we just need to make sure the vjp dtype matches the `data` dtype in the corresponding DataArray. For this component I think it's a float. (So yea it might have been wrong before?)